### ISSUE TITLE
Improve Error type

### ISSUE THREAD

**Issue by aleokdev** [ggez/ggez#1021]
Right now, (almost) all error variants of `GameError` have no actual information about the error that a program can parse except a string. End programs wanting to know the exact issue that happened (for example, to solve it) cannot do so.

The actual error string can be provided in the display implementation, as is common in most crates. This'll also reduce duplicate code in the crate.

Another issue is that error variant fields are very undescriptive, for example `ResourceNotFound(String, Vec<(std::path::PathBuf, GameError)>),`; I suggest we change the tuple variants for named member ones.

**Comment by aleokdev** [comment]
Another note: Since `GameError` is the library error type, I don't think it's appropriate for users to wrap other error types in it; it doesn't make much sense to do so. I suggest removing `GameError::CustomError` altogether. Users wanting to output other error types within an event handler can set the `E` type parameter to a generic error type.

**Comment by aleokdev** [comment]
I can work on this once #1020 is merged with the main branch.

**Comment by aleokdev** [comment]
Second note: After further inspection, I noticed that there is currently an issue with the EventHandler error type parameter:
```rs
pub trait EventHandler<E = GameError>
where
    E: std::error::Error,
```
As far as I know, there is currently no way to actually use a generic error type here. `Box<dyn std::error::Error>` won't work, and `anyhow::Error` won't either, as neither implement the Error trait. As far as I can tell, there is actually no reason to have this limitation, and any type as `E` should work as I cannot see any place where the error trait is actually required.

**Comment by aleokdev** [comment]
@nobbele This issue is not fixed yet; The main problem right now is that error variants only contain Strings. See the first comment for more info.

**Comment by nobbele** [comment]
oh this was a different issue, right

**Comment by aleokdev** [comment]
Working on this now that the wgpu branch is merged.
